pkg/controller/podautoscaler/metrics/metrics_client.go

```diff
@@ -137,23 +158,30 @@ func (h *HeapsterMetricsClient) GetCpuConsumptionAndRequestInMillis(namespace st
 			}
 		}
 	}
-	if len(podNames) == 0 && len(podList.Items) > 0 {
-		return 0, 0, 0, time.Time{}, fmt.Errorf("no running pods")
+	if readyPods.Len() == 0 {
+		return 0, 0, nil, fmt.Errorf("no running and ready pods")
 	}
 	if missing || requestSum == 0 {
-		return 0, 0, 0, time.Time{}, fmt.Errorf("some pods do not have request for cpu")
+		return 0, 0, nil, fmt.Errorf("some pods do not have request for cpu")
```

Comment:
I guess that is not what we want to do. Ready&running missing pods should be treated as 0 for scale-up and as 100% for scale down. Please see https://github.com/kubernetes/kubernetes/issues/30471#issuecomment-249879455.